src/server/src/main/java/io/cassandrareaper/ReaperApplicationConfiguration.java

```diff
@@ -14,6 +14,8 @@
 
 package io.cassandrareaper;
 
+import io.cassandrareaper.core.Cluster;
```

Comment:
the config classes should not depend upon the model beans.
this relates to the work in https://github.com/thelastpickle/cassandra-reaper/issues/163
specifically the last comment https://github.com/thelastpickle/cassandra-reaper/issues/163#issuecomment-325483387 where it's visualised that this class is one of the untangled classes (and it would be nice to keep it that way).

i think this has to be solved by moving the logic introduced in `getJmxCredentialsForCluster(..)` back up a level.
This concern (lack of encapsulation) is also repeated in the concern that we've added a parameter to the _connect_ methods in `JmxConnectionFactory` (procedural vs object-oriented code).

I'm aware of the difficulty around the jmx connections (and credentials) now being dependant on both host _and cluster_. My first guess is, to keep the whole `Map<String, JmxCredentials> jmxCredentials` as a final instance field in `JmxConnectionFactory` since this is not call dynamic. And then replace  each `String host` parameter in the different _connect*_ methods with a `Node node` parameter, where Node is a pojo with cluster and host.

The default `connectionTimeout` could also be stored as a final instance field, allowing simpler _connect*_ methods to be added and used in most places. But that's outside the scope of this PR.